proposals/NNNN-sharing-wifi-ssid-and-password.md

```diff
@@ -0,0 +1,169 @@
+
+```json
+{
+  "tcpIpAddress": "192.168.1.1",
+  "tcpPort": 12345,
+  "WifiSSID":"InCarWiFiAP",
+  "wifiPassword":"12345678",
+  "wifiSecurity":"WIFI_SECURITY_WPA2"
+}
+```
+
+On receiving `Transport Event Update` with WiFi credentials, Proxy should check if the mobile device is already connected to WiFi access point mentioned in `Transport Event Update`. If the mobile device is not already connected, it should try connecting to the WiFi access point. 
+
+On successful connection or if the mobile device is already connected to WiFi access point mentioned in `Transport Event Update`, core and proxy should follow use case described in a proposal [Supporting simultaneous multiple transports](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0141-multiple-transports.md "Supporting simultaneous multiple transports").
+
+
+## Potential downsides
+
+None
+
+## Impact on existing code
+
+1. Implementation of HMI API described above in Core. 
+2. Android App will need to include WiFi permissions in Manifest file if they support WiFi as secondary transport.
+3. Network Capability needs to be added in iOS App.
+4. Implementation in proxy to connect WiFi.
+
+### Sample Code for WiFi Connection in Android
+
+Below code can be used to connect to Wi-Fi Access point if SSID and password are known.
+
+```
+    private void connectToWiFi(String ssid,String password){
+        WifiConfiguration conf = new WifiConfiguration();
+        conf.SSID = "\"" + ssid + "\"";
+        conf.preSharedKey = "\""+ password +"\"";
+
+        m_wifiManager.addNetwork(conf);
+        List<WifiConfiguration> list = m_wifiManager.getConfiguredNetworks();
+        for( WifiConfiguration i : list ) {
+            if(i.SSID != null && i.SSID.equals("\"" + ssid + "\"")) {
+                m_wifiManager.disconnect();
+                m_wifiManager.enableNetwork(i.networkId, true);
+                m_wifiManager.reconnect();
+                break;
+            }
+        }
+    }
+```
+Permissions required to use above mentioned API's are as follows:
+
+```
+<uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
+<uses-permission android:name="android.permission.CHANGE_WIFI_STATE" />
+```
+### Sample Code for WiFi Connection in iOS
+
+Below code can be used to connect to Wi-Fi Access point if SSID and password are known.
+
+```
+    NEHotspotConfiguration * config = [[NEHotspotConfiguration alloc] initWithSSID:ssid passphrase:password isWEP:NO];
+    [config setJoinOnce:NO];
+    
+    NEHotspotConfigurationManager* configManager = [NEHotspotConfigurationManager sharedManager];
+    [configManager applyConfiguration:config completionHandler:^(NSError * _Nullable error) {
+        [self showAlert:[error localizedDescription]];
+        NSLog(@"Error %@",[error debugDescription]);
+    }];
+```
+Capabilities required to use above mentioned API's are as follows:
+
+```
+Hotspot Configuration
+```
+  
+
+## Alternatives considered
+
+None

```

Comment:
If possible, can you please complete this section? Per the PR template:
>Describe alternative approaches to addressing the same problem, and why you chose this approach instead.

Reply:
Added now.